File: ios/Packages/SolennixCore/Sources/SolennixCore/Route.swift
import Foundation

// MARK: - Route

/// All navigable destinations in the app. Used with NavigationPath / .navigationDestination(for:).
public enum Route: Hashable {

    // MARK: Events
    case eventList
    case eventDetail(id: String)
    case eventForm(id: String? = nil, clientId: String? = nil, date: Date? = nil)
    case eventChecklist(id: String)
    case eventFinances(id: String)
    case eventPayments(id: String)
    case eventProducts(id: String)
    case eventExtras(id: String)
    case eventSupplies(id: String)
    case eventEquipment(id: String)
    case eventShoppingList(id: String)
    case eventPhotos(id: String)
    case eventContractPreview(id: String)
    case eventStaff(id: String)

    // Clients
    case clientList
    case clientDetail(id: String)
    case clientForm(id: String? = nil)
    case quickQuote

    // MARK: Staff (Personal / Colaboradores)
    case staffList
    case staffDetail(id: String)
    case staffForm(id: String? = nil)

    // MARK: Staff Teams (Cuadrillas)
    case staffTeamList
    case staffTeamDetail(id: String)
    case staffTeamForm(id: String? = nil)

    // MARK: Products
    case productList
    case productDetail(id: String)
    case productForm(id: String? = nil)

    // MARK: Inventory
    case inventoryList
    case inventoryDetail(id: String)
    case inventoryForm(id: String? = nil)

    // MARK: Tools
    case search(query: String = "")
    case settings

    // MARK: Event Form Links
    case eventFormLinks

    // MARK: Settings Sub-Routes
    case editProfile
    case changePassword
    case businessSettings
    case contractDefaults
    case pricing
    case subscription
    case notificationPreferences
    case about
}

File: android/app/src/main/java/com/creapolis/solennix/ui/navigation/Route.kt
package com.creapolis.solennix.ui.navigation

import kotlinx.serialization.Serializable

@Serializable
sealed class Route {
    @Serializable data class EventDetail(val id: String) : Route()
    @Serializable data class EventForm(val id: String? = null, val clientId: String? = null) : Route()
    @Serializable data class EventChecklist(val id: String) : Route()
    @Serializable data class ClientDetail(val id: String) : Route()
    @Serializable data class ClientForm(val id: String? = null) : Route()
    @Serializable data class QuickQuote(val clientId: String? = null) : Route()
    @Serializable data class ProductDetail(val id: String) : Route()
    @Serializable data class ProductForm(val id: String? = null) : Route()
    @Serializable data class InventoryDetail(val id: String) : Route()
    @Serializable data class InventoryForm(val id: String? = null) : Route()
    @Serializable data object Onboarding : Route()
    @Serializable data object EditProfile : Route()
    @Serializable data object ChangePassword : Route()
    @Serializable data object BusinessSettings : Route()
    @Serializable data object ContractDefaults : Route()
    @Serializable data object Pricing : Route()
    @Serializable data object About : Route()
    @Serializable data object Privacy : Route()
    @Serializable data object Terms : Route()
    @Serializable data object EventFormLinks : Route()
}
